tracer/test/Datadog.Trace.TestHelpers/RemoteConfigTestHelper.cs

```diff
@@ -67,7 +67,7 @@ internal static async Task<GetRcmRequest> WaitRcmRequestAndReturnLast(this MockT
                 {
                     // prefer a request that has been processed, meaning the config states have been filled in
                     request = JsonConvert.DeserializeObject<GetRcmRequest>(lastRemoteConfigPayload);
-                    if (request?.Client?.State?.ConfigStates?.Count is not null and > 0)
+                    if (request?.Client?.State?.ConfigStates?.Count is > 0 && (appliedServiceNames == null || appliedServiceNames.All(s => request.Client.State.ConfigStates.Any(c => c.Id == s))))
```

Comment:
This is the only change in this file? The rest is reformatting?

Reply:
mhh also removed the `await Task.Delay(1000);` in `SetupRcmAndWait`, added a datetime on the output that helped debugging, and made a const of timeout for `WaitRcm`

Reply:
ha sorry misread in this class, yes, the only change is the added parameter and the conditions to match all ids before returning

Reply:
and yes, sorry for the reformating, still need to adjust my ctrl+k+d  ... or not use it at all 🥲